src/libImaging/GifEncode.c

```diff
@@ -233,6 +233,13 @@ ImagingGifEncode(Imaging im, ImagingCodecState state, UINT8* buf, int bytes)
                         }
 
                 }
+                /* Potential special case for xsize==1 */
+                if (state->x < state->xsize) {
+                    this = state->buffer[state->x++];
+                } else {
+                    EMIT_RUN(label0);
+                    break;
+                }
 
                 this = state->buffer[state->x++];
```

Comment:
```suggestion
```

I don't have valgrind setup, so I'm not on top of this - but did you mean to remove this line (so that it could be replaced by your new ones)? Because doing so causes the tests to pass.

Reply:
Yeah, that was a bad rebase, as this was originally (accidentally) written on top of a branch that was a year old. (I mean, last update was Dec 23, how out of date could it be?)

Reply:
You should have an invite from a couple days back for a docker image that will do valgrind.

Reply:
Ah, yes, I see. Thanks